File: guet/steps/check/git_required_check.py
from guet.git.git import Git
from guet.steps.check.check import Check

GIT_REQUIRED_MESSAGE = "Git not installed in this directory."


class GitRequiredCheck(Check):

    def __init__(self, git: Git):
        super().__init__(GIT_REQUIRED_MESSAGE)
        self.git = git

    def should_stop(self, args):
        try:
            # This depends on GitProxy loading it's Git internally, causing the FileNotFoundError.
            self.git.hooks_present()
            return False
        except FileNotFoundError:
            return True

File: src/steps/check/gitRequiredCheck.ts
import { Check } from './check';
import { Args } from '../../args';
import { filesystem } from 'gluegun';

export class GitRequiredCheck extends Check {
    constructor() {
        super('No git directory in this folder.');
    }

    public shouldStop(args: Args): boolean {
        return (
            filesystem.exists(filesystem.path(filesystem.cwd(), '.git')) && true
        );
    }
}
